src/main/java/bio/terra/service/common/gcs/GcsUriUtils.java

```diff
@@ -0,0 +1,49 @@
+package bio.terra.service.common.gcs;
+
+import com.google.cloud.storage.BlobId;
+
+public final class GcsUriUtils {
+  private static final String GS_BUCKET_PATTERN = "[a-z0-9_.\\-]{3,222}";
+
+  /**
+   * Parse a Google Cloud Storage URI into its component pieces
+   *
+   * @param uri of type gs://<bucket_name>/<file_path_inside_bucket>
+   * @return Object representing uri pieces
+   */
+  public static BlobId parseBlobUri(String uri) {
```

Comment:
Like `validateBlobUri()`, this also throws
```suggestion
  public static BlobId parseBlobUri(String uri) throws IllegalArgumentException {
```